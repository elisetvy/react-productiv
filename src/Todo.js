import React from "react";

/** Simple presentation component for a todo.
 *
 * Props:
 * - todo: like { id, title, description, priority }
 *
 * { EditableTodo, TopTodo } -> Todo
 **/

function Todo({ id, title, description, priority } ) {
  console.log("TODO TITLE IS: ", title);
  const testTitle = title;
  console.log(testTitle);
  return (
      <div className="Todo" id={id}>
        {/* <p>{testTitle}</p> */}
        <div><b>{title}</b> <small>(priority: {priority})</small></div>
        <div><small>{description}</small></div>
      </div>
  );
}

export default Todo;
