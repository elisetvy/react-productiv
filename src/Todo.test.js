import React from "react";
import { render } from "@testing-library/react";
import Todo from "./Todo";

describe("Todo component", function (){
  const initialTodo= {
    id: 1,
    title: "Code!",
    description: "Write some code",
    priority: 2,
  }

  test("Should render without crashing", function () {
    render(<Todo todo={initialTodo} />);
  });

  test("contains expected title", function () {
    const result = render(<Todo todo={initialTodo} />);
    expect(result.queryByText("Code!")).toBeInTheDocument();
  });

  test("passes snapshot", function (){
    const { container } = render(<Todo todo={initialTodo} />);
    expect(container).toMatchSnapshot();
  });
});