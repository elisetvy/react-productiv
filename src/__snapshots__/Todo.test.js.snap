// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component passes snapshot 1`] = `
<div>
  <div
    class="Todo"
    id="1"
  >
    <div>
      <b>
        Code!
      </b>
       
      <small>
        (priority: 
        2
        )
      </small>
    </div>
    <div>
      <small>
        Write some code
      </small>
    </div>
  </div>
</div>
`;
